packages/open-truss/src/configuration/engine-v1.tsx

```diff
@@ -1,33 +1,46 @@
 import React from 'react'
-import type { YamlObject, YamlType } from '@/utils/yaml'
+import { YamlObjectShape, YamlShape } from '../utils/yaml'
 import { type RenderingEngine, type ReactTree, type COMPONENTS } from './apply'
+import { z } from 'zod'
 
-type Components = JSX.Element
+type Components = React.JSX.Element
 
-export interface BaseOpenTrussComponentV1Props {
-  key?: number
-  children?: Components
-  data: YamlType
-  config: WorkflowV1
+const FrameBase = z.object({
+  view: z.object({
+    component: z.string(),
+    props: YamlObjectShape,
+  }),
+  data: z.string(),
```

Comment:
I think this will change to an object but don't know what the shape will be yet.

Reply:
Oops yea this should be `YamlShape`! It was a string early on when I first started this but was fixed recently. I'll update then merge. Thanks for the review!